File: cw/binary/dialog.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import base

import cw


class Dialog(base.CWBinaryBase):
    """台詞データ"""
    def __init__(self, parent, f, yadodata=False):
        base.CWBinaryBase.__init__(self, parent, f, yadodata)
        self.coupons = f.string(True)
        self.text = f.string(True)

        self.data = None

    def get_data(self):
        if self.data is None:
            self.data = cw.data.make_element("Dialog")
            e = cw.data.make_element("RequiredCoupons", self.coupons)
            self.data.append(e)
            e = cw.data.make_element("Text", self.text)
            self.data.append(e)
        return self.data

    @staticmethod
    def unconv(f, data):
        coupons = ""
        text = ""

        for e in data:
            if e.tag == "RequiredCoupons":
                coupons = e.text
            elif e.tag == "Text":
                text = e.text

        f.write_string(coupons, True)
        f.write_string(text, True)

def main():
    pass

if __name__ == "__main__":
    main()

File: cw/binary/dialog.cs
//!/usr/bin/env python
// -*- coding: utf-8 -*-

import base;

import cw;


class Dialog : base.CWBinaryBase {
    // """台詞データ""";
    public Dialog(UNK parent, UNK f, bool yadodata=false) : base(parent, f, yadodata) {
        this.coupons = f.string(true);
        this.text = f.string(true);

        this.data = null;
    }

    public UNK get_data() {
        if (this.data == null) {
            this.data = cw.data.make_element("Dialog");
            e = cw.data.make_element("RequiredCoupons", this.coupons);
            this.data.append(e);
            e = cw.data.make_element("Text", this.text);
            this.data.append(e);
        }
        return this.data;
    }

    public static void unconv(UNK f, UNK data) {
        coupons = "";
        text = "";

        foreach (var e in data) {
            if (e.tag == "RequiredCoupons") {
                coupons = e.text;
            } else if (e.tag == "Text") {
                text = e.text;
            }
        }

        f.write_string(coupons, true);
        f.write_string(text, true);
    }
}
